torch_disc/torch_disc/csrc/disc_jit.cpp

```diff
@@ -161,29 +147,95 @@ std::string MhloConversaion(const std::shared_ptr<Graph>& graph) {
   std::ofstream outfile(in_fname);
   outfile << parsable_mlir << std::endl;
   outfile.flush();
-  return in_fname;
+  return std::make_tuple(in_fname, input_dev_str, output_dev_str);
 }
 
-void CallDiscCompiler(const std::string& mlir_fname) {
+std::string CallDiscCompiler(const std::string& mlir_fname) {
   std::string out_fname = mlir_fname + ".out";
   std::string cmd = DiscCMD(mlir_fname, out_fname);
   TORCH_CHECK(std::system(cmd.c_str()) == 0,
               "disc compile failed with cmd: " + cmd);
+  return out_fname;
 }
 
-void DiscCompilation(const std::shared_ptr<Graph>& graph) {
-  for (auto node : graph->nodes()) {
-    if (node->kind() == prim::FusionGroup) {
-      auto sub_graph = node->g(attr::Subgraph);
-      std::string fname = MhloConversaion(sub_graph);
-      CallDiscCompiler(fname);
+std::vector<const torch::jit::Value*> ToConstValue(
+    c10::ArrayRef<torch::jit::Value*> values) {
+  std::vector<const torch::jit::Value*> const_vals;
+  std::copy(values.begin(), values.end(), std::back_inserter(const_vals));
+  return const_vals;
+}
+
+std::vector<c10::IValue> DiscCompilation(const std::shared_ptr<Graph>& graph) {
```

Comment:
Please add some documentation about what has been done in this function, since it's long.

Reply:
I have done some code refactoring, please take a look at the latest commit, thanks!